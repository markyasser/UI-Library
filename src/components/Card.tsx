import React, { useState } from 'react';

interface CardProps {
  title: string;
  content: string;
  showFooter?: boolean; // Optional prop to control footer visibility
}

const Card: React.FC<CardProps> = ({ title, content, showFooter = true }) => {
  const [likes, setLikes] = useState(0); // Additional state not covered by the test

  return (
    <div style={{ border: '1px solid #ddd', padding: '1rem', borderRadius: '5px' }}>
      <h3>{title}</h3>
      <p>{content}</p>

      {/* Logic to show/hide footer */}
      {showFooter && (
        <div style={{ marginTop: '1rem', borderTop: '1px solid #ccc', paddingTop: '0.5rem' }}>
          <button onClick={() => setLikes(likes + 1)}>Like</button>
          <span>{likes} Likes</span>
        </div>
      )}
    </div>
  );
};

export default Card;
