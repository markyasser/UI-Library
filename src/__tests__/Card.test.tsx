import React from 'react';
import { render } from '@testing-library/react';
import Card from '../components/Card';

describe('Card Component', () => {
  test('renders without crashing', () => {
    const { container } = render(<Card title="Test Title" content="Test Content" />);
    expect(container).toBeInTheDocument();
  });
});
