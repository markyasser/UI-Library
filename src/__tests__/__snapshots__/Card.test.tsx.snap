// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="border: 1px solid #ddd; padding: 1rem; border-radius: 5px;"
  >
    <h3>
      Test Title
    </h3>
    <p>
      Test Content
    </p>
    <div
      style="margin-top: 1rem; border-top: 1px solid #ccc; padding-top: 0.5rem;"
    >
      <button>
        Like
      </button>
      <span>
        0 Likes
      </span>
    </div>
  </div>
</DocumentFragment>
`;
